src/prometheus_server.py

```diff
@@ -34,13 +37,25 @@ def reload_configuration(self) -> bool:
           True if reload succeeded (returned 200 OK); False otherwise.
         """
         url = f"{self.base_url}/-/reload"
```

Comment:
iirc, @mmanciop encourages to use SIGHUP (container.exec kill -HUP?) instead of http api, for DoS reasons.

Reply:
Not really DoS, but it is less complex. Although it is tricky to validate whether it worked.

Reply:
Well, I did actually write that Pebble log streamer…

Reply:
If we can detect the outcome of the reload very reliably with it, it is worth considering.

Reply:
> If we can detect the outcome of the reload very reliably with it, it is worth considering.

Is this possible @rbarry82 ? If so, do we have a charm implementing this?